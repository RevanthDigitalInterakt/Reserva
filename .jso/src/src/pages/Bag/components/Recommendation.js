  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.Recommendation = Recommendation;
  var _slicedToArray2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[2]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[3]);
  var _EventProvider = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[4]));
  var _useRecommendation2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[5]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function Recommendation() {
    var _useRecommendation = (0, _useRecommendation2.default)(),
      products = _useRecommendation.products,
      showSection = _useRecommendation.showSection,
      setProducts = _useRecommendation.setProducts,
      setShowSection = _useRecommendation.setShowSection;
    var _useApolloFetchPolicy = (0, _$$_REQUIRE(_dependencyMap[6]).useApolloFetchPolicyStore)(['getFetchPolicyPerKey']),
      getFetchPolicyPerKey = _useApolloFetchPolicy.getFetchPolicyPerKey;
    var _useProductRecommenda = (0, _$$_REQUIRE(_dependencyMap[7]).useProductRecommendationsLazyQuery)({
        context: {
          clientName: 'gateway'
        },
        fetchPolicy: getFetchPolicyPerKey('productRecommendations')
      }),
      _useProductRecommenda2 = (0, _slicedToArray2.default)(_useProductRecommenda, 1),
      getProductRecommendation = _useProductRecommenda2[0];
    var onToggleSection = (0, _react.useCallback)(function () {
      return setShowSection(!showSection);
    }, [setShowSection, showSection]);
    (0, _react.useEffect)(function () {
      getProductRecommendation().then(function (_ref) {
        var data = _ref.data;
        if (data) {
          setProducts(data == null ? undefined : data.productRecommendations);
        }
      });
    }, [getProductRecommendation, setProducts]);
    (0, _react.useEffect)(function () {
      if (!products || !(products != null && products.length)) return;
      if (!_EventProvider.default) return;
      try {
        var newItems = products.map(function (item) {
          var _item$priceRange, _item$priceRange$sell, _item$categoryTree;
          return {
            price: item == null ? undefined : (_item$priceRange = item.priceRange) == null ? undefined : (_item$priceRange$sell = _item$priceRange.sellingPrice) == null ? undefined : _item$priceRange$sell.lowPrice,
            item_id: item == null ? undefined : item.productId,
            item_name: item == null ? undefined : item.productName,
            item_category: Object.values(item == null ? undefined : (_item$categoryTree = item.categoryTree) == null ? undefined : _item$categoryTree.map(function (i) {
              return i.href;
            })).join('|')
          };
        });
        _EventProvider.default.logEvent('page_view', {
          item_brand: _$$_REQUIRE(_dependencyMap[8]).defaultBrand.picapau
        });
        _EventProvider.default.logEvent('view_item_list', {
          items: newItems,
          item_brand: (0, _$$_REQUIRE(_dependencyMap[9]).getBrandByUrl)(products)
        });
      } catch (error) {
        _$$_REQUIRE(_dependencyMap[10]).ExceptionProvider.captureException(error);
      }
    }, [products]);
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsxs)(_$$_REQUIRE(_dependencyMap[11]).Fragment, {
      children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[12]).Box, {
        paddingX: "xxxs",
        children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[13]).Divider, {
          marginTop: "xs",
          variant: "fullWidth"
        })
      }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[12]).Box, {
        testID: "com.usereserva:id/products_recomendation_list",
        paddingLeft: "micro",
        children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[14]).Button, {
          onPress: onToggleSection,
          hitSlop: {
            left: 50,
            top: 15,
            bottom: 15
          },
          flexDirection: "row",
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsxs)(_$$_REQUIRE(_dependencyMap[12]).Box, {
            flexDirection: "row",
            alignItems: "flex-start",
            marginY: "xxs",
            marginBottom: "xxs",
            children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[12]).Box, {
              marginRight: "micro",
              children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[15]).IconLegacy, {
                name: "Handbag",
                size: 20
              })
            }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[12]).Box, {
              flex: 1,
              children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[16]).Typography, {
                variant: "subtituloSessoes",
                children: "Outros produtos que voc\xEA pode gostar"
              })
            }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[12]).Box, {
              marginRight: "md",
              children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[15]).IconLegacy, {
                name: "ArrowDown",
                color: "preto",
                size: "20",
                style: showSection ? {
                  transform: [{
                    rotate: '-180deg'
                  }, {
                    translateY: 4
                  }]
                } : {
                  transform: [{
                    translateY: 8
                  }]
                }
              })
            })]
          })
        })
      }), !!showSection && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsxs)(_$$_REQUIRE(_dependencyMap[11]).Fragment, {
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[12]).Box, {
          paddingX: "xxxs",
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[13]).Divider, {
            marginBottom: "xs",
            variant: "fullWidth"
          })
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[12]).Box, {
          mt: "quarck",
          paddingX: "micro",
          children: products && (products == null ? undefined : products.length) > 0 && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_reactNative.View, {
            style: {
              marginBottom: 20
            },
            children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[11]).jsx)(_$$_REQUIRE(_dependencyMap[17]).ListHorizontalProducts, {
              horizontal: true,
              products: products
            })
          })
        })]
      })]
    });
  }
