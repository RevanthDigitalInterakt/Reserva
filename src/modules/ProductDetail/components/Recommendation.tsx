import { useLazyQuery } from '@apollo/client';
import {
  Box, Button, Divider, Icon, Typography,
} from '@usereservaapp/reserva-ui';
import React, { useEffect, useState } from 'react';
import * as Animatable from 'react-native-animatable';
import { createAnimatableComponent } from 'react-native-animatable';
import { productSearch } from '../../../graphql/products/productSearch';
import { ListHorizontalProducts } from './ListHorizontalProducts';
import EventProvider from '../../../utils/EventProvider';

interface RecommendationProps {
  handleScrollToTheTop?: () => void;
}

export const Recommendation = ({
  handleScrollToTheTop,
}: RecommendationProps) => {
  const [skip, setSkip] = useState(false);
  const pageSize = 36;
  const [showMore, setShowMore] = useState(true);
  const [products, setProducts] = useState<any>([]);

  const BoxAnimated = createAnimatableComponent(Box);

  const [getProductData] = useLazyQuery(productSearch, {
    // skip,
    variables: {
      skusFilter: 'ALL_AVAILABLE',
      hideUnavailableItems: true,
      orderBy: 'OrderByTopSaleDESC',
      to: pageSize - 1,
      simulationBehavior: 'default',
      productOriginVtex: false,
      salesChannel: '4',
    },
    fetchPolicy: 'no-cache',
    nextFetchPolicy: 'no-cache',
  });

  const saveItems = async (items: any) => {
    const arrayProductsId = items.map((elem) => elem.productId);

    const arrayWithoutDuplicates = items.filter((element, index) => index === arrayProductsId.indexOf(element.productId));

    setProducts(arrayWithoutDuplicates.slice(0, 6));
  };

  useEffect(() => {
    const handleSearch = async () => {
      const { data, loading } = await getProductData();

      if (!loading) {
        await saveItems(data.productSearch.products);
      }
    };

    handleSearch();
  }, []);

  useEffect(() => {
    if (!products || !products?.length) return;
    if (!EventProvider) return;

    try {
      const newItems = products.map((item) => ({
        price: item?.priceRange?.sellingPrice?.lowPrice,
        item_id: item?.productId,
        item_name: item?.productName,
        item_category: Object.values(item?.categoryTree.map((i) => (i.href))).join('|'),
      }));
      EventProvider.logEvent('view_item_list', {
        items: newItems,
      });
    } catch (error) {
      EventProvider.captureException(error);
    }
  }, [products, EventProvider]);

  return (
    <>
      <Box>
        <Divider marginTop="xs" variant="fullWidth" />
      </Box>

      <Box paddingLeft="quarck" marginTop="xxxs" marginBottom="xxxs">
        <Button
          onPress={() => setShowMore(!showMore)}
          hitSlop={{ left: 50, top: 15, bottom: 15 }}
          flexDirection="row"
        >
          <BoxAnimated
            flexDirection="row"
            alignItems="flex-start"
            marginY="xxs"
            marginBottom="xxs"
          >
            <Box marginRight="micro">
              <Icon name="Handbag" size={20} />
            </Box>

            <Box flex={1}>
              <Typography variant="subtituloSessoes">
                Outros produtos que vocÃª pode gostar
              </Typography>
            </Box>
            <Box marginRight="md">
              <Icon
                name={showMore ? 'ArrowDown' : 'ArrowDown'}
                style={
                  showMore
                    ? { transform: [{ rotate: '-180deg' }, { translateY: 4 }] }
                    : { transform: [{ translateY: 8 }] }
                }
                color="preto"
                size={showMore ? '20' : '20'}
              />
            </Box>
          </BoxAnimated>
        </Button>
      </Box>
      {showMore ? (
        <>
          <Box>
            <Divider marginBottom="xs" variant="fullWidth" />
          </Box>
          <Box mt="quarck">
            {products && products?.length > 0 && (
              <Animatable.View animation="fadeIn" style={{ marginBottom: 20 }}>
                <ListHorizontalProducts
                  horizontal
                  products={products || []}
                  handleScrollToTheTop={handleScrollToTheTop}
                />
              </Animatable.View>
            )}
          </Box>
          <Box>
            <Divider marginBottom="xs" variant="fullWidth" />
          </Box>
        </>
      ) : (
        <Box>
          <Divider marginBottom="xs" variant="fullWidth" />
        </Box>
      )}
    </>
  );
};
