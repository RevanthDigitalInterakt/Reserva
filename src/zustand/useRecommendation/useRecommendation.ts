import { create } from 'zustand';
import type { ProductOutput } from '../../base/graphql/generated';
import type { ProductQL } from '../../graphql/products/productSearch';
import type { IRecommendationState } from './types/recommendation';

const parseProductsGatewayToVtex = (products: ProductOutput[]) => {
  const vtexProducts: ProductQL[] = products.map((product) => ({
    ...product,
    skuSpecifications: [],
    categoryTree: product.categoryTree?.map((category) => ({
      name: category,
    })),
    items: product.items.map((item) => ({
      ...item,
      images: item.images.map((image) => ({
        imageUrl: image,
      })),
      sellers: item.sellers.map((seller) => ({
        ...seller,
        commertialOffer: {
          Price: seller?.commertialOffer?.price,
          Installments: seller?.commertialOffer?.installments.map((installment) => ({
            Value: installment?.value,
            TotalValuePlusInterestRate: installment?.totalValuePlusInterestRate,
            NumberOfInstallments: installment?.numberOfInstallments,
          })) || [],
          ListPrice: seller?.commertialOffer?.listPrice,
          PriceWithoutDiscount: seller?.commertialOffer?.priceWithoutDiscount,
          AvailableQuantity: seller?.commertialOffer?.availableQuantity,
          Tax: seller?.commertialOffer?.tax,
          discountHighlights: [],
          teasers: [],
          spotPrice: seller.commertialOffer?.spotPrice,
          taxPercentage: seller.commertialOffer?.taxPercentage,
        },
      })),
    })),
  })) as ProductQL[];
  return vtexProducts;
};

const useRecommendation = create<IRecommendationState>((set) => ({
  showMore: true,
  products: [],
  setShowMore: (value: boolean) => set(() => ({ showMore: value })),
  setProducts: (value: Array<ProductOutput>) => set(
    () => ({ products: parseProductsGatewayToVtex(value) }),
  ),
}));
export default useRecommendation;
