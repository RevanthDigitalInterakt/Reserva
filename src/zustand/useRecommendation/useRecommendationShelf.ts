import { create } from 'zustand';
import {
  RecommendationShelfDocument, TrackProvidersEnum,
  type RecommendationShelfQuery, type RecommendationShelfQueryVariables,
  SmarthintShelfChannelEnum, TrackPageTypeEnum,
} from '../../base/graphql/generated';
import type { IRecommendationShelfState } from './types/recommendationShelf';
import { getApolloClient } from '../../utils/getApolloClient';

const useRecommendationShelf = create<IRecommendationShelfState>(() => ({
  onSearchShelf: async (_page: string) => {
    try {
      const client = getApolloClient();

      const { data } = await client
        .query<RecommendationShelfQuery, RecommendationShelfQueryVariables>({
        context: { clientName: 'gateway' },
        query: RecommendationShelfDocument,
        variables: {
          input: {
            providers: [TrackProvidersEnum.Smarthint],
            userIdentifier: 'email@gmail.com',
            channel: [SmarthintShelfChannelEnum.Padrao],
            pageType: TrackPageTypeEnum.Home,
            position: 2,
          },
        },
        fetchPolicy: 'no-cache',
      });
      return [data.recommendationShelf];
    } catch (e) {
      return [];
    }
  },
}));

export default useRecommendationShelf;
