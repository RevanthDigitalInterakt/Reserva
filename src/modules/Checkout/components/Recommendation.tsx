import {
  Box, Button, Divider, Icon, Typography,
} from '@usereservaapp/reserva-ui';
import React, { useEffect } from 'react';
import * as Animatable from 'react-native-animatable';
import { createAnimatableComponent } from 'react-native-animatable';
import { ListHorizontalProducts } from './ListHorizontalProducts';
import EventProvider from '../../../utils/EventProvider';
import useRecommendation from '../../../zustand/useRecommendation/useRecommendation';
import { useProductRecommendationsLazyQuery } from '../../../base/graphql/generated';

export function Recommendation() {
  const {
    products, showMore, setProducts, setShowMore,
  } = useRecommendation();

  const BoxAnimated = createAnimatableComponent(Box);

  const [getProductRecommendation] = useProductRecommendationsLazyQuery({
    context: {
      clientName: 'gateway',
    },
  });

  useEffect(() => {
    getProductRecommendation().then(({ data }) => {
      if (data) {
        setProducts(data?.productRecommendations);
      }
    });
  }, [getProductRecommendation, setProducts]);

  useEffect(() => {
    if (!products || !products?.length) return;
    if (!EventProvider) return;
    try {
      const newItems = products.map((item) => ({
        price: item?.priceRange?.sellingPrice?.lowPrice,
        item_id: item?.productId,
        item_name: item?.productName,
        item_category: Object.values(item?.categoryTree?.map((i) => (i.href))).join('|'),
      }));
      EventProvider.logEvent('view_item_list', {
        items: newItems,
      });
    } catch (error) {
      EventProvider.captureException(error);
    }
  }, [products]);

  return (
    <>
      <Box paddingX="xxxs">
        <Divider marginTop="xs" variant="fullWidth" />
      </Box>
      <Box testID="com.usereserva:id/products_recomendation_list" paddingLeft="micro">
        <Button
          onPress={() => setShowMore(!showMore)}
          hitSlop={{ left: 50, top: 15, bottom: 15 }}
          flexDirection="row"
        >
          <BoxAnimated
            flexDirection="row"
            alignItems="flex-start"
            marginY="xxs"
            marginBottom="xxs"
          >
            <Box marginRight="micro">
              <Icon name="Handbag" size={20} />
            </Box>
            <Box flex={1}>
              <Typography variant="subtituloSessoes">
                Outros produtos que vocÃª pode gostar
              </Typography>
            </Box>
            <Box marginRight="md">
              <Icon
                name={showMore ? 'ArrowDown' : 'ArrowDown'}
                style={
                  showMore
                    ? { transform: [{ rotate: '-180deg' }, { translateY: 4 }] }
                    : { transform: [{ translateY: 8 }] }
                }
                color="preto"
                size={showMore ? '20' : '20'}
              />
            </Box>
          </BoxAnimated>
        </Button>
      </Box>
      {!!showMore && (
        <>
          <Box paddingX="xxxs">
            <Divider marginBottom="xs" variant="fullWidth" />
          </Box>
          <Box mt="quarck" paddingX="micro">
            {products && products?.length > 0 && (
              <Animatable.View animation="fadeIn" style={{ marginBottom: 20 }}>
                <ListHorizontalProducts
                  horizontal
                  products={products}
                />
              </Animatable.View>
            )}
          </Box>
        </>
      )}
    </>
  );
}
