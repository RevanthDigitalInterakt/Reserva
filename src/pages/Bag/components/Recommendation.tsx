import React, { useCallback, useEffect } from 'react';
import { View } from 'react-native';

import { useProductRecommendationsLazyQuery } from '../../../base/graphql/generated';
import { ExceptionProvider } from '../../../base/providers/ExceptionProvider';
import { Box } from '../../../components/Box/Box';
import { Button } from '../../../components/Button';
import { Divider } from '../../../components/Divider/Divider';
import { IconLegacy } from '../../../components/IconLegacy/IconLegacy';
import { Typography } from '../../../components/Typography/Typography';
import { ListHorizontalProducts } from '../../../modules/Checkout/components/ListHorizontalProducts';
import EventProvider from '../../../utils/EventProvider';
import { defaultBrand } from '../../../utils/defaultWBrand';
import { getBrandByUrl } from '../../../utils/getBrandByURL';
import { useApolloFetchPolicyStore } from '../../../zustand/useApolloFetchPolicyStore';
import useRecommendation from '../../../zustand/useRecommendation/useRecommendation';

export function Recommendation() {
  const {
    products, showSection, setProducts, setShowSection,
  } = useRecommendation();

  const { getFetchPolicyPerKey } = useApolloFetchPolicyStore(['getFetchPolicyPerKey']);

  const [getProductRecommendation] = useProductRecommendationsLazyQuery({
    context: { clientName: 'gateway' },
    fetchPolicy: getFetchPolicyPerKey('productRecommendations'),
  });

  const onToggleSection = useCallback(() => (
    setShowSection(!showSection)
  ), [setShowSection, showSection]);

  useEffect(() => {
    getProductRecommendation().then(({ data }) => {
      if (data) {
        setProducts(data?.productRecommendations);
      }
    });
  }, [getProductRecommendation, setProducts]);

  useEffect(() => {
    if (!products || !products?.length) return;
    if (!EventProvider) return;
    try {
      const newItems = products.map((item) => ({
        price: item?.priceRange?.sellingPrice?.lowPrice,
        item_id: item?.productId,
        item_name: item?.productName,
        item_category: Object.values(item?.categoryTree?.map((i) => (i.href))).join('|'),
      }));
      EventProvider.logEvent('page_view', {
        wbrand: defaultBrand.picapau,
      });
      EventProvider.logEvent('view_item_list', {
        items: newItems,
        wbrand: getBrandByUrl(products),
      });
    } catch (error) {
      ExceptionProvider.captureException(error);
    }
  }, [products]);

  return (
    <>
      <Box paddingX="xxxs">
        <Divider marginTop="xs" variant="fullWidth" />
      </Box>
      <Box testID="com.usereserva:id/products_recomendation_list" paddingLeft="micro">
        <Button
          onPress={onToggleSection}
          hitSlop={{ left: 50, top: 15, bottom: 15 }}
          flexDirection="row"
        >
          <Box
            flexDirection="row"
            alignItems="flex-start"
            marginY="xxs"
            marginBottom="xxs"
          >
            <Box marginRight="micro">
              <IconLegacy name="Handbag" size={20} />
            </Box>

            <Box flex={1}>
              <Typography variant="subtituloSessoes">
                Outros produtos que vocÃª pode gostar
              </Typography>
            </Box>
            <Box marginRight="md">
              <IconLegacy
                name="ArrowDown"
                color="preto"
                size="20"
                style={
                  showSection
                    ? { transform: [{ rotate: '-180deg' }, { translateY: 4 }] }
                    : { transform: [{ translateY: 8 }] }
                }
              />
            </Box>
          </Box>
        </Button>
      </Box>
      {!!showSection && (
        <>
          <Box paddingX="xxxs">
            <Divider marginBottom="xs" variant="fullWidth" />
          </Box>
          <Box mt="quarck" paddingX="micro">
            {products && products?.length > 0 && (
              <View style={{ marginBottom: 20 }}>
                <ListHorizontalProducts
                  horizontal
                  products={products}
                />
              </View>
            )}
          </Box>
        </>
      )}
    </>
  );
}
