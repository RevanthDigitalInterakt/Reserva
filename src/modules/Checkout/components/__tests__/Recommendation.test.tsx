import { MockedProvider } from '@apollo/client/testing';
import {
  act,
  render, screen,
} from '@testing-library/react-native';
import { theme } from '@usereservaapp/reserva-ui';
import type { DocumentNode } from 'graphql';
import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { renderHook } from '@testing-library/react-hooks';
import { ProductRecommendationsQuery, ProductRecommendationsDocument } from '../../../../base/graphql/generated';
import { Recommendation } from '../Recommendation';
import { gatewayProductsListMock } from '../../../../zustand/useRecommendation/mocks/productsList';
import useRecommendation from '../../../../zustand/useRecommendation/useRecommendation';

interface IApolloMock<T> {
  request: {
    query: DocumentNode,
    variables: object,
  },
  result: {
    data: T
  }
}

const apolloMocks: Array<IApolloMock<ProductRecommendationsQuery>> = [
  {
    request: {
      query: ProductRecommendationsDocument,
      variables: {},
    },
    result: {
      data: {
        productRecommendations: gatewayProductsListMock,
      },
    },
  },
];

describe('Test Component Recommendation', () => {
  beforeEach(async () => {
    jest.clearAllMocks();
  });

  describe('Testing Recommendation', () => {
    it('renders without error and match snapshot', async () => {
      const Wrapper = (
        <ThemeProvider theme={theme}>
          <MockedProvider mocks={apolloMocks} addTypename={false}>
            <Recommendation />
          </MockedProvider>
        </ThemeProvider>
      );

      render(Wrapper);

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should make a successful call to getProductRecommendation and add the products to zustand', async () => {
      const { result } = renderHook(() => useRecommendation());

      const Wrapper = (
        <ThemeProvider theme={theme}>
          <MockedProvider mocks={apolloMocks} addTypename={false}>
            <Recommendation />
          </MockedProvider>
        </ThemeProvider>
      );

      render(Wrapper);

      expect(result.current.products.length).toEqual(0);

      await act(async () => {
        await screen.rerender(Wrapper);
      });

      expect(result.current.products.length).toEqual(6);
    });
  });
});
