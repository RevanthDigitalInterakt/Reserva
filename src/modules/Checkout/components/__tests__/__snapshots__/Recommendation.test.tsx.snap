// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component Recommendation Testing Recommendation renders without error and match snapshot 1`] = `
Array [
  <View
    paddingX="xxxs"
    style={
      Array [
        Object {
          "paddingLeft": 16,
          "paddingRight": 16,
        },
      ]
    }
  >
    <View
      backgroundColor="divider"
      height={1}
      marginTop="xs"
      style={
        Array [
          Object {
            "backgroundColor": "#D8D9DA",
            "height": 1,
            "marginTop": 32,
          },
        ]
      }
      variant="fullWidth"
    />
  </View>,
  <View
    paddingLeft="micro"
    style={
      Array [
        Object {
          "paddingLeft": 12,
        },
      ]
    }
    testID="com.usereserva:id/products_recomendation_list"
  >
    <View
      alignItems="center"
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 15,
            "left": 50,
            "top": 15,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          alignItems="flex-start"
          flexDirection="row"
          marginBottom="xxs"
          marginY="xxs"
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginBottom": 24,
                "marginTop": 24,
              },
              Object {},
            ]
          }
        >
          <View
            marginRight="micro"
            style={
              Array [
                Object {
                  "marginRight": 12,
                },
              ]
            }
          >
            <RNSVGSvgView
              bbHeight={20}
              bbWidth={20}
              color="preto"
              focusable={false}
              size={20}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Array [
                    Object {
                      "color": "#333333",
                      "height": 20,
                      "width": 20,
                    },
                  ],
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  focusable={false}
                  id="Logo"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Array [
                        Object {},
                      ],
                      Object {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={19.477}
                  vbWidth={16.569}
                >
                  <RNSVGGroup>
                    <RNSVGGroup
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0.5,
                          0.5,
                        ]
                      }
                      name="Sacola"
                    >
                      <RNSVGPath
                        d="M5.388,7.8,4.5,19.909a1.73,1.73,0,0,0,1.73,1.73H18.339a1.73,1.73,0,0,0,1.73-1.73L18.913,7.8Z"
                        fill={null}
                        matrix={
                          Array [
                            1,
                            0,
                            0,
                            1,
                            -4.5,
                            -3.162,
                          ]
                        }
                        name="Caminho_65"
                        propList={
                          Array [
                            "fill",
                            "stroke",
                            "strokeLinecap",
                            "strokeLinejoin",
                          ]
                        }
                        stroke={4281545523}
                        strokeLinecap={1}
                        strokeLinejoin={1}
                      />
                      <RNSVGPath
                        d="M4.5,9H17.9"
                        fill={null}
                        matrix={
                          Array [
                            1,
                            0,
                            0,
                            1,
                            -3.489,
                            -4.362,
                          ]
                        }
                        name="Caminho_66"
                        propList={
                          Array [
                            "fill",
                            "stroke",
                            "strokeLinecap",
                            "strokeLinejoin",
                          ]
                        }
                        stroke={4281545523}
                        strokeLinecap={1}
                        strokeLinejoin={1}
                      />
                      <RNSVGPath
                        d="M18.362,18.181a3.181,3.181,0,0,0-6.362,0"
                        fill={null}
                        matrix={
                          Array [
                            1,
                            0,
                            0,
                            1,
                            -7.612,
                            -15,
                          ]
                        }
                        name="Caminho_67"
                        propList={
                          Array [
                            "fill",
                            "stroke",
                            "strokeLinecap",
                            "strokeLinejoin",
                          ]
                        }
                        stroke={4281545523}
                        strokeLinecap={1}
                        strokeLinejoin={1}
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            flex={1}
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "ReservaSerif-Regular",
                    "fontSize": 20,
                  },
                ]
              }
              variant="subtituloSessoes"
            >
              Outros produtos que vocÃª pode gostar
            </Text>
          </View>
          <View
            marginRight="md"
            style={
              Array [
                Object {
                  "marginRight": 48,
                },
              ]
            }
          >
            <RNSVGSvgView
              bbHeight={20}
              bbWidth={20}
              color="preto"
              focusable={false}
              size="20"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Array [
                    Object {
                      "color": "#333333",
                      "height": 20,
                      "width": 20,
                    },
                    Object {
                      "transform": Array [
                        Object {
                          "rotate": "-180deg",
                        },
                        Object {
                          "translateY": 4,
                        },
                      ],
                    },
                  ],
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
            >
              <RNSVGGroup
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  focusable={false}
                  height="100%"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Array [
                        Object {
                          "height": "100%",
                          "width": "100%",
                        },
                      ],
                      Object {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={19.477}
                  vbWidth={16.569}
                  width="100%"
                >
                  <RNSVGGroup>
                    <RNSVGGroup
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0.5,
                          0.5,
                        ]
                      }
                      name="Arrow_-_icon"
                    >
                      <RNSVGPath
                        d="M.873,13.174,0,12.3,5.719,6.593,0,.873.873,0l6.58,6.593L4.99,9.057Z"
                        fill={4281545523}
                        matrix={
                          Array [
                            6.123233995736766e-17,
                            1,
                            -1,
                            6.123233995736766e-17,
                            13.174,
                            0,
                          ]
                        }
                        name="right001_-_E012"
                        propList={
                          Array [
                            "fill",
                            "stroke",
                          ]
                        }
                        stroke={4281545523}
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    paddingX="xxxs"
    style={
      Array [
        Object {
          "paddingLeft": 16,
          "paddingRight": 16,
        },
      ]
    }
  >
    <View
      backgroundColor="divider"
      height={1}
      marginBottom="xs"
      style={
        Array [
          Object {
            "backgroundColor": "#D8D9DA",
            "height": 1,
            "marginBottom": 32,
          },
        ]
      }
      variant="fullWidth"
    />
  </View>,
  <View
    mt="quarck"
    paddingX="micro"
    style={
      Array [
        Object {
          "marginTop": 4,
          "paddingLeft": 12,
          "paddingRight": 12,
        },
      ]
    }
  />,
]
`;
